documentation/src/main/asciidoc/topics/con_file_based_cache_stores.adoc

```diff
@@ -0,0 +1,77 @@
+[id='file-stores_{context}']
+= File-Based Cache Stores
+
+File-based cache stores provide persistent storage on the local host filesystem where {brandname} is running.
+For clustered caches, file-based cache stores are unique to each {brandname} node.
+
+You should use file-based cache stores to persist data that overflows from memory when the data container exceeds size or time restrictions.
+
+[WARNING]
+====
+Never use filesystem-based cache stores on shared file systems, such as an NFS or Samba share, because they do not provide file locking capabilities and data corruption can occur.
+
+Additionally if you attempt to use transactional caches with shared file systems, unrecoverable failures can happen when writing to files during the commit phase.
+====
+
+[discrete]
+== Soft-Index File Stores
+
+[NOTE]
+====
+`SoftIndexFileStore` is the default implementation for file-based cache stores.
+====
+
+`SoftIndexFileStore` is a Java implementation that uses a variant of **B+
+Tree** that is cached in-memory using Java soft references.
+The **B+ Tree**, called `Index` is offloaded on the file system to a single file that is purged and rebuilt each time the cache store restarts.
```

Comment:
The index is actually stored in a user configured amount of indices. The default is 16. There are two reasons for this number, one is that concurrent writes can have the index cause a bottleneck. And the second which is more important now is that when iterating over the soft index all entries are read from a single index, thus we want to minimize how many entries can be read at the same time.